CHANGELOG.md

```diff
@@ -5,6 +5,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Added ReleaseBuffer() function and call it with "defer". Also removed
```

Comment:
This changelog is aimed at end-users, so this detail is a little too much. As Alex suggested, I think "Fixed TChannel memory leak that would occur during server-side errors." is simple. Maybe replace "leak" with "memory pressure" or something